### ISSUE TITLE
Unable to find parts git repository

### ISSUE THREAD

**Issue by deshipu** [fritzing/fritzing-app#3890]
## Current Behaviour

Fritzing can't find the parts repository. No parts are loaded.

![Screenshot_2021-10-14_11-23-15](https://user-images.githubusercontent.com/1181357/137289733-e670d937-6bfe-4e1d-8619-1650ed2d69e6.png)

**Build:**
Version 0.9.9

**Operating System:**
Linux Mint

**Steps to reproduce:**
- Download the appimage
- Make it executable
- Start it

## Expected Behaviour
Fritizng should load the parts definitions.

**Comment by KjellMorgenstern** [comment]
Hello @deshipu, thanks for the report.
The AppImage includes the most recent part database, and there should be nothing to update. A simple guess would be that Fritzing gets confused by a previous installation. Did you manually trigger the db regeneration, or did it start by itself?

**Comment by vanepp** [comment]
You could try Help->Enable debug log which will open a debug log window (if the particular error that is occurring has debug code enabled in it an error message should appear in the debug window.) Another thing to try is clearing your user directories like this:

There are two user directories (with your parts and the parts database) which don't get touched during an install (to not affect your sketches during upgrades). On Windows they are in

c:\users\username\AppData\Fritzing\roaming\Fritzing (which is a hidden directory so you need to enable hidden directories in explorer) and

c:\Users\username\My Documents\Fritzing (where username is your windows id)

If you don't have any parts or sketches you want to keep you can just delete those two directories and Fritzing will recreate them, or you can move them aside by renaming them if you want to keep something in them.

linux

~/Documents/Fritzing/parts
~/.config/Fritzing 

Mac

/Users/username/Documents/Fritzing/parts
~/.config/Fritzing

**Comment by chris-hatton** [comment]
I'm not so sure this should be closed @deshipu - I am facing the same issue, but creating a whole new Linux user to work around Fritzing's apparent inability to upgrade from older versions is not really acceptable.

**Comment by KjellMorgenstern** [comment]
So you installed 0.9.3b from the ubuntu package on linux mint? Which version of mint? And did you also install the fritzing-parts package?

**Comment by mrprole** [comment]
I managed to solve the problem after this operations and all work well. Some strange behavior with autoremove.
mrprole@mrprole-Swift-SFX14-41G:~$ sudo apt remove fritzing-parts
[sudo] password for mrprole: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  fritzing-data libqt5serialport5
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  fritzing fritzing-parts
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
After this operation, 150 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
mrprole@mrprole-Swift-SFX14-41G:~$ sudo apt autoremove
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
mrprole@mrprole-Swift-SFX14-41G:~$ sudo apt remove fritzing-parts
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  fritzing-data libqt5serialport5
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  fritzing fritzing-parts
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
After this operation, 150 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 572595 files and directories currently installed.)
Removing fritzing (0.9.3b+dfsg-10build2) ...
Removing fritzing-parts (0.9.3b-5) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for mailcap (3.69ubuntu1) ...
Processing triggers for desktop-file-utils (0.26-1ubuntu2) ...
mrprole@mrprole-Swift-SFX14-41G:~$ sudo apt autoremove
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages will be REMOVED:
  fritzing-data libqt5serialport5
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
After this operation, 7 409 kB disk space will be freed.
Do you want to continue? [Y/n] y

**Comment by benreu** [comment]
I also run Linux Mint (19 & 20). I have the same problem. 

When I uninstall the _fritzing-data_ package using apt or equivalent, Fritzing 0.9.9 starts working. When I reinstall _fritzing-data_, Fritzing 0.9.9 complains that it is unable to find parts git repository. The _fritzing-parts_ package doesn't affect the Fritzing 0.9.9 parts repository, installed or uninstalled.

This problem irritated me the longest time, I didn't know the fix until today. Every time a new release of Fritzing came out, I would donate some more of my hard earned money, only to run into the same problem. I commented on a previous issue, only to have it closed with the explanation that the problem no longer is relevant. It was a pretty disappointing experience. 

I like Fritzing and have used it commercially for 10+ years, but this was almost enough to make me look somewhere else. I will say, I'm glad the community opened this issue and found a workaround.

**Comment by benreu** [comment]
One more thing I found:

If I open Fritzing 0.9.9, wait until it shows the error `unable to find parts git repository`, then uninstall _fritzing-data_, and then click OK in Fritzing, Fritzing will try to load a new sketch and hang at 14%.

![Screenshot from 2022-01-11 16-42-26](https://user-images.githubusercontent.com/24444193/149026333-dabc09c9-d219-4fe3-a36c-a6225a85c0ca.png)

**Comment by KjellMorgenstern** [comment]
Uninstalling the fritzing-parts package does still leave a directory "/usr/share/fritzing". This directory needs to be manually removed, Fritzing is still confused by it.

**Comment by KjellMorgenstern** [comment]
@benreu The issue was not closed by Fritzing, but the original poster.

**Comment by benreu** [comment]
> Uninstalling the fritzing-parts package does still leave a directory "/usr/share/fritzing". This directory needs to be manually removed, Fritzing is still confused by it.

If you use Synaptic and select "Mark for complete removal" on `fritzing-parts` and `fritzing-data` it will remove directory "/usr/share/fritzing".

**Comment by benreu** [comment]
> @benreu The issue was not closed by Fritzing, but the original poster.

No, it was an older issue about parts directory not found. It doesn't matter any more, as the solution has now been found for Linux Mint. I'm assuming this issue is only Linux Mint related, as nobody else is complaining. The previous issue was affecting all Linux users, if I remember correctly.

**Comment by KjellMorgenstern** [comment]
Ah, yes, there were other issues, for example Windows defender would scan all files during the first start, Fritzing would therefore slow down and be "This apllication is not responding" for ten minutes or more. When the user then kills Fritzing, the part database would be invalid... those and similar issues are solved.
Anyway, i reopened the issue, i agree manually researching and then removing the old configuration should not be needed. 


If possible, Fritzing should ignore those files. If in doubt, Fritzing should show an error message that helps the user to decide.

**Comment by christianbaun** [comment]
> Uninstalling the fritzing-parts package does still leave a directory "/usr/share/fritzing". This directory needs to be manually removed, Fritzing is still confused by it.

I had the same issue in Debian 11 these steps solved it:

```
sudo apt remove fritzing-parts
sudo rm -rf /usr/share/fritzing/
```

**Comment by benreu** [comment]
There must be a difference between Debian and Mint, since I uninstalled Fritzing on 2 different Mint computers, and I did not have to remove the _/usr/share/fritzing/_ directory manually.

**Comment by deshipu** [comment]
For me, on Mint, the culprit was actually in `~/.local/share/fritzing` – I suppose some earlier install must have put stuff in there.

**Comment by GeniuScientist** [comment]
Windows user here!
So, I'm having this error when I try to build Fritzing (with Qt Creator). I already checked My Documents and APPDATA, but, since this is my first time building it, there were no folders. I might be missing something obvious, I would appreciate if someone helps me.

**Comment by ead0601** [comment]
I had the same issue, and this is what I did to resolve it.

sudo apt remove --purge fritzing
sudo apt remove --purge fritzing-parts
sudo apt remove --purge fritzing-data

only after purging --purge, it start to work with the user based install at ~/bin/fritzing....

**Comment by benreu** [comment]
@ead0601 Please list you OS / distro. This problem seems to affect certain distros more than others

**Comment by deshipu** [comment]
I wonder if it would make sense to remove all the external search paths for this when making the appimage.

**Comment by KjellMorgenstern** [comment]
All good, next release won't have the issue.

**Comment by ead0601** [comment]
> @ead0601 Please list you OS / distro. This problem seems to affect certain distros more than others

Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal

**Comment by failiz** [comment]
Was this fixed in 0.9.10? or will it be fixed in the next version?

**Comment by KjellMorgenstern** [comment]
Summary of the discussion above: We have found that 3rd party installs of Fritzing 0.9.3 can cause the issue observed. A left over part library folder would confuse Fritzing, and it would try to load the parts from there. This has been fixed in 0.9.10, so the workaround with apt remove --purge is not needed anymore. I think we can close this issue.

**Comment by jiapei100** [comment]
Issue continues of today... Dec 22, 2023.... 

I built **fritzing-app** with **Qt-6.6.1**.

I'm thinking there **MUST** be a way from within **Fritzing IDE** to specify the root/directory of [fritzing-parts](https://github.com/fritzing/fritzing-parts) ???
